addons/website_event_track/controllers/event_track.py

```diff
@@ -392,30 +392,58 @@ def event_track_proposal(self, event, **post):
 
     @http.route(['''/event/<model("event.event"):event>/track_proposal/post'''], type='http', auth="public", methods=['POST'], website=True)
     def event_track_proposal_post(self, event, **post):
-        tags = []
-        for tag in event.allowed_track_tag_ids:
-            if post.get('tag_' + str(tag.id)):
-                tags.append(tag.id)
+        if not event.can_access_from_current_website():
+            raise NotFound()
+
+        #  Only accept existing tag indices.
+        input_tag_indices = [int(tag_id) for tag_id in post['tags'].split(',') if tag_id]
+        valid_tag_indices = request.env['event.track.tag'].search([('id', 'in', input_tag_indices)]).ids
+
+        #  Contact name is required. Therefore, empty contacts are not considered here. At least one of contact_phone
+        #  and contact_email must be filled. No contact is created if the user do not tick the additional information
+        #  checkbox on the track proposal form.
+        contact = request.env['res.partner']
+        if 'add_contact_information' in post:
+            valid_contact_emails = tools.email_split(post['contact_email'])
+            valid_contact_email = valid_contact_emails[0] if valid_contact_emails else False,
+            if valid_contact_email or post['contact_phone']:
+                contact = request.env['res.partner'].sudo().create({
+                    'name': post['contact_name'],
+                    'email': valid_contact_email,
+                    'phone': post['contact_phone'],
+                })
+            else:
+                raise exceptions.ValidationError(_("Format Error : please enter a valid contact phone or contact email."))
```

Comment:
Here I tried to enforce condition with a Validation if not valid contact mean exist. 

However the idea of a "valid phone" is quite tricky. Indeed, as discussed with lna, there is no reason to accept only the phone formatted according to the geoip of current user. If that is the only contact mean, it is hard to know if it is valid as there are a lot of different possible formats. This could be quite restrictive for the user.

This could be maybe be later improved with ongoing work in 
https://www.odoo.com/web#action=4043&cids=1&id=1865062&menu_id=4720&model=project.task&view_type=form
with a widget allowing to choose country and the according format.

Maybe we can accept any phone, or if needed go to a mandatory contact email only.

Reply:
Indeed I could subscribe to OXP with my belgian phone number while being in another country. I think this comment could be added as a comment and maybe in commit message.